thirdparty/hnswlib/hnswlib/hnswalg.h

```diff
@@ -600,24 +610,29 @@ class HierarchicalNSW : public AlgorithmInterface<dist_t> {
     }
 
     void
-    loadIndex(const std::string& location, SpaceInterface<dist_t>* s, size_t max_elements_i = 0) {
-        std::ifstream input(location, std::ios::binary);
-
-        if (!input.is_open())
-            throw std::runtime_error("Cannot open file");
+    loadIndex(const std::string_view location, const knowhere::LoadConfig& config, size_t max_elements_i = 0) {
+        auto input = knowhere::FileReader(location, true);
 
-        // get file size:
-        input.seekg(0, input.end);
-        std::streampos total_filesize = input.tellg();
-        input.seekg(0, input.beg);
+        size_t dim;
+        readBinaryPOD(input, metric_type_);
+        readBinaryPOD(input, data_size_);
+        readBinaryPOD(input, dim);
+        if (metric_type_ == 0) {
+            space_ = new hnswlib::L2Space(dim);
+        } else if (metric_type_ == 1) {
+            space_ = new hnswlib::InnerProductSpace(dim);
+        } else {
+            throw std::runtime_error("Invalid metric type " + std::to_string(metric_type_));
+        }
```

Comment:
What schema is this. Looks like we change the schema from what HNSW generate to something else?

Reply:
Just decode from the index file, like `loadIndex(knowhere::MemoryIOReader& input, size_t max_elements_i = 0)`